From 22699996b13e803cb7aab2f6058d489375c9754f Mon Sep 17 00:00:00 2001
From: Joe Thornber <ejt@redhat.com>
Date: Mon, 29 Mar 2021 16:34:03 +0100
Subject: [PATCH 4381/4585] [Backport] dm btree spine: remove paranoid
 node_check call in node_prep_for_write()

mainline inclusion
from mainline-v5.13-rc1
commit f73e2e70ec48c9a9d45494c4866230a5059062ad
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5JCAH

Reference: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?id=f73e2e70ec48c9a9d45494c4866230a5059062ad

--------------------------------

Remove this extra BUG_ON() that calls node_check() -- which avoids extra crc checking.

Signed-off-by: Joe Thornber <ejt@redhat.com>
Signed-off-by: Mike Snitzer <snitzer@redhat.com>
Signed-off-by: Zhang Changzhong <zhangchangzhong@huawei.com>
---
 drivers/md/persistent-data/dm-btree-spine.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/md/persistent-data/dm-btree-spine.c b/drivers/md/persistent-data/dm-btree-spine.c
index e03cb9e48773..c4b386b2be97 100644
--- a/drivers/md/persistent-data/dm-btree-spine.c
+++ b/drivers/md/persistent-data/dm-btree-spine.c
@@ -30,8 +30,6 @@ static void node_prepare_for_write(struct dm_block_validator *v,
 	h->csum = cpu_to_le32(dm_bm_checksum(&h->flags,
 					     block_size - sizeof(__le32),
 					     BTREE_CSUM_XOR));
-
-	BUG_ON(node_check(v, b, 4096));
 }
 
 static int node_check(struct dm_block_validator *v,
-- 
2.17.1

